src/AutoRest.CSharp/Common/Output/Builders/MultipartSerializationMethodsBuilder.cs

```diff
@@ -0,0 +1,243 @@
+                    "SerializeMultipart",
+                    null,
+                    null,
+                    MethodSignatureModifiers.Private,
+                    typeof(BinaryData),
+                    null,
+                    new Parameter[] { KnownParameters.Serializations.Options }),
+                BuildMultipartSerializationMethodBody().ToArray());
+
+            /* internal virtual MultipartFormDataRequestContent ToMultipartRequestContent() */
+            yield return new Method(
+                new MethodSignature(
+                    Configuration.ApiTypes.ToMultipartRequestContentName,
+                    null,
+                    null,
+                    MethodSignatureModifiers.Internal | MethodSignatureModifiers.Virtual,
+                    MultipartFormDataRequestContentProvider.Instance.Type,
+                    null,
+                    Array.Empty<Parameter>()),
+                BuildToMultipartRequestContentMethodBody(multipart).ToArray());
+        }
+        public static SwitchCase BuildMultipartWriteSwitchCase(MultipartObjectSerialization multipart, ModelReaderWriterOptionsExpression options)
+        {
+            return new SwitchCase(
+                Serializations.MultipartFormat,
+                Return(new InvokeInstanceMethodExpression(null, "SerializeMultipart", new[] { options }, null, false)));
+        }
+
+        public static IEnumerable<MethodBodyStatement> BuildMultipartSerializationMethodBody()
+        {
+            var serializeCallExpression = new InvokeInstanceMethodExpression(null, Configuration.ApiTypes.ToMultipartRequestContentName, Array.Empty<ValueExpression>(), null, false);
+            return new MethodBodyStatement[]
+                        {
+                            UsingDeclare("content", MultipartFormDataRequestContentProvider.Instance.Type, serializeCallExpression, out var content),
+                            /*using MemoryStream stream = new MemoryStream();*/
+                            UsingDeclare("stream", typeof(MemoryStream), New.Instance(typeof(MemoryStream), Array.Empty<ValueExpression>()), out var stream),
+                            /*content.WriteTo(stream, cancellationToken);*/
+                            MultipartFormDataRequestContentProvider.Instance.WriteTo(content, stream, null),
+                            new IfElseStatement(GreaterThan(stream.Property("Position"), IntExpression.MaxValue),
+                            /*return BinaryData.FromStream(stream);*/
+                            Return(BinaryDataExpression.FromStream(stream, false)),
+                            /*return new BinaryData(stream.GetBuffer().AsMemory(0, (int)stream.Position));*/
+                            Return(New.Instance(typeof(BinaryData), new[]{ new InvokeInstanceMethodExpression((new StreamExpression(stream)).GetBuffer, "AsMemory", new[] { Literal(0), new CastExpression(stream.Property("Position"), typeof(int)) }, null, false) }))
```

Comment:
`MemoryStream.GetBuffer` is a method, therefore we should make this expression as a method.
Also because `stream.GetBuffer()` returns `byte[]`, and `AsMemory` should be an extension method, we should use the `InvokeStaticMethodExpression` with `CallAsExtension: true` here to ensure the namespace of this extension is properly included.

Reply:
Also New.Isntance could take a params of ValueExpression therefore the array here is unnecessary and could make it more clear

```suggestion
                            Return(New.Instance(typeof(BinaryData), new InvokeStaticMethodExpression(typeof(MemoryExtensions), nameof(MemoryExtensions.AsMemory), new ValueExpression[] { stream.GetBuffer(), Literal(0), stream.Position.CastTo(typeof(int)) }, CallAsExtension: true)))
```